<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Measuring gene expression divergence between gene pairs • exdiva</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Measuring gene expression divergence between gene pairs">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">exdiva</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/exdiva_vignette.html">Measuring gene expression divergence between gene pairs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/almeidasilvaf/exdiva/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Measuring gene expression divergence between
gene pairs</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      <p>VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent,
Belgium</p>
<br><h4 data-toc-skip class="author">Yves
Van de Peer</h4>
            <address class="author_afil">
      <p>VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent, Belgium</p>
<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/exdiva/blob/devel/vignettes/exdiva_vignette.Rmd" class="external-link"><code>vignettes/exdiva_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>exdiva_vignette.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Gene and genome duplications are important sources of evolutionary
innovation. At the transcriptional level, some genes can diverge in
expression after duplication, eventually leading to novel functions,
while others under stronger selection pressures preserve expression
levels. To understand the evolution of gene expression following
duplication, we developed <em><a href="https://github.com/exdiva" class="external-link">exdiva</a></em>, a package that
provides users with a set of metrics to compute expression divergence
between duplicate pairs. <em><a href="https://github.com/exdiva" class="external-link">exdiva</a></em> integrates with the
Bioconductor ecosystem by accepting as input core S4 classes, including
<code>SummarizedExperiment</code> objects (for bulk RNA-seq data),
<code>SingleCellExperiment</code> objects (for single-cell RNA-seq
data), and <code>SpatialExperiment</code> objects (for spatial
transcriptomics data).</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <em><a href="https://github.com/exdiva" class="external-link">exdiva</a></em> from Bioconductor with
the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"exdiva"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Then, you can load the package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/exdiva" class="external-link">exdiva</a></span><span class="op">)</span></span></code></pre></div>
<p>We will also load other required packages to use in this
vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># for reproducibility</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>In this vignette, we will use data from a publicly available
experiment (SRA Study ID: SRP213876) in which authors investigated seed
germination in <em>A. thaliana</em> seeds exposed to red and far-red
light <span class="citation">(Tognacca et al. 2019)</span>. Expression
data in transcripts per million (TPM) were obtained from <a href="https://www.refine.bio/" class="external-link">refine.bio</a> and stored in a
<code>SummarizedExperiment</code> object. Let’s first take a look at
them.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load expression data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ath_se</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show `SummarizedExperiment` object</span></span>
<span><span class="va">ath_se</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 22545 6 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): TPM</span></span>
<span><span class="co">#&gt; rownames(22545): AT1G01010 AT1G01020 ... ATMG01310 ATMG01410</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(6): SRR9659514 SRR9659515 ... SRR9659518 SRR9659519</span></span>
<span><span class="co">#&gt; colData names(3): experiment tissue treatment</span></span>
<span></span>
<span><span class="co"># Take a look at assay and colData slots</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ath_se</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SRR9659514 SRR9659515 SRR9659516 SRR9659517 SRR9659518 SRR9659519</span></span>
<span><span class="co">#&gt; AT1G01010   77.71092   56.03711   76.01148   88.05801   70.60007   85.64420</span></span>
<span><span class="co">#&gt; AT1G01020  460.21158  417.35464  464.21402  247.83946  360.69977  303.03580</span></span>
<span><span class="co">#&gt; AT1G01030   49.03193   27.52164   46.13850   15.80170   21.04727   38.64825</span></span>
<span><span class="co">#&gt; AT1G01040  564.81950  380.74603  750.09094  572.19684  863.37067  687.85156</span></span>
<span><span class="co">#&gt; AT1G01050  605.36630  526.64270  666.37146  477.44717  684.85050  637.83575</span></span>
<span><span class="co">#&gt; AT1G01060  731.03310  603.77970  898.30460  103.11177  149.65310  123.45633</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ath_se</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 3 columns</span></span>
<span><span class="co">#&gt;             experiment      tissue   treatment</span></span>
<span><span class="co">#&gt;            &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; SRR9659514   SRP213876       seeds         Red</span></span>
<span><span class="co">#&gt; SRR9659515   SRP213876       seeds         Red</span></span>
<span><span class="co">#&gt; SRR9659516   SRP213876       seeds         Red</span></span>
<span><span class="co">#&gt; SRR9659517   SRP213876       seeds     Far-red</span></span>
<span><span class="co">#&gt; SRR9659518   SRP213876       seeds     Far-red</span></span>
<span><span class="co">#&gt; SRR9659519   SRP213876       seeds     Far-red</span></span></code></pre></div>
<p>The entire set of duplicated gene pairs in the <em>A. thaliana</em>
genome was obtained with the Bioconductor package <em><a href="https://bioconductor.org/packages/3.19/doubletrouble" class="external-link">doubletrouble</a></em>
<span class="citation">(Almeida-Silva and Van de Peer 2024)</span>. Data
are stored in a data frame with 3 columns as below:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ath_dups</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Take a look at the first rows of the data frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ath_dups</span><span class="op">)</span></span>
<span><span class="co">#&gt;        dup1      dup2 type</span></span>
<span><span class="co">#&gt; 1 AT2G17930 AT4G36080   SD</span></span>
<span><span class="co">#&gt; 2 AT1G03060 AT4G02660   SD</span></span>
<span><span class="co">#&gt; 3 AT2G45540 AT3G60920   SD</span></span>
<span><span class="co">#&gt; 4 ATCG01280 ATCG00860   SD</span></span>
<span><span class="co">#&gt; 5 AT3G28770 AT5G39870   SD</span></span>
<span><span class="co">#&gt; 6 AT1G05570 AT2G31960   SD</span></span></code></pre></div>
<p>To demonstrate metrics that require gene coexpression networks, we
used <em><a href="https://bioconductor.org/packages/3.19/BioNERO" class="external-link">BioNERO</a></em>
to infer a signed coexpression network with data from <span class="citation">Wan et al. (2019)</span>, available on <a href="https://www.refine.bio/" class="external-link">refine.bio</a>. Then, we saved the data
frame of module assignment and module eigengenes in the objects
below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load and show data frame of genes per module</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">genes_and_modules</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genes_and_modules</span><span class="op">)</span></span>
<span><span class="co">#&gt;       Genes        Modules</span></span>
<span><span class="co">#&gt; 1 AT1G01040         orange</span></span>
<span><span class="co">#&gt; 2 AT1G01050    greenyellow</span></span>
<span><span class="co">#&gt; 3 AT1G01060    darkorange2</span></span>
<span><span class="co">#&gt; 4 AT1G01080    lightyellow</span></span>
<span><span class="co">#&gt; 5 AT1G01090    darkorange2</span></span>
<span><span class="co">#&gt; 6 AT1G01100 darkolivegreen</span></span>
<span></span>
<span><span class="co"># Load and show data frame of module eigengenes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ME</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ME</span><span class="op">)</span></span>
<span><span class="co">#&gt;            MEdarkmagenta     MEplum2 MEdarkolivegreen MEgreenyellow     MEplum1</span></span>
<span><span class="co">#&gt; SRR9331156   -0.10867620 -0.09945584     -0.183336028    0.04141757  0.04071353</span></span>
<span><span class="co">#&gt; SRR9331157    0.03272060 -0.03500161      0.007757575   -0.06353141 -0.04264297</span></span>
<span><span class="co">#&gt; SRR9331158   -0.08398072 -0.06468694      0.081085373    0.06774658  0.06892207</span></span>
<span><span class="co">#&gt; SRR9331159    0.07944359  0.11759883      0.092786208    0.02838570  0.08992221</span></span>
<span><span class="co">#&gt; SRR9331160    0.05883327 -0.02752882      0.013886315   -0.11184364 -0.04331340</span></span>
<span><span class="co">#&gt; SRR9331161    0.23683993  0.13817651     -0.239434683   -0.06339785  0.01315021</span></span>
<span><span class="co">#&gt;            MElightyellow      MEcyan    MEgrey60 MElightgreen      MEblack</span></span>
<span><span class="co">#&gt; SRR9331156   0.005871434  0.28156756  0.12547552 -0.059923496  0.098952397</span></span>
<span><span class="co">#&gt; SRR9331157  -0.031379476 -0.08983925 -0.07475956 -0.233099121  0.050483212</span></span>
<span><span class="co">#&gt; SRR9331158  -0.129102982 -0.18508131 -0.14166572  0.018012265  0.061365626</span></span>
<span><span class="co">#&gt; SRR9331159  -0.125119712 -0.04581348 -0.06248737  0.009834189 -0.210741559</span></span>
<span><span class="co">#&gt; SRR9331160   0.006980476 -0.01652390  0.01593942 -0.091500946  0.006723972</span></span>
<span><span class="co">#&gt; SRR9331161  -0.086003591  0.11185457 -0.01823782 -0.216059157 -0.080836641</span></span>
<span><span class="co">#&gt;                MEwhite    MEorange MEdarkorange2 MEmediumorchid    MEpurple</span></span>
<span><span class="co">#&gt; SRR9331156  0.08887598  0.20141331    0.08322848     0.01545135 -0.02562527</span></span>
<span><span class="co">#&gt; SRR9331157 -0.04970812 -0.05168774   -0.08969168     0.05588269  0.07066997</span></span>
<span><span class="co">#&gt; SRR9331158  0.15228640  0.12202822    0.11355630     0.20633328  0.08858222</span></span>
<span><span class="co">#&gt; SRR9331159 -0.11736622 -0.11017594   -0.06766365    -0.19711635  0.09052339</span></span>
<span><span class="co">#&gt; SRR9331160  0.03354560  0.07372503   -0.07013520    -0.05175000  0.15381676</span></span>
<span><span class="co">#&gt; SRR9331161 -0.22868743 -0.06243383   -0.23518730    -0.18128462  0.14940833</span></span>
<span><span class="co">#&gt;               MEcoral1    MEsalmon MEdarkorange  MEthistle2 MEpaleturquoise</span></span>
<span><span class="co">#&gt; SRR9331156 -0.15034983 -0.17455918   0.17308273  0.06892352      0.15856487</span></span>
<span><span class="co">#&gt; SRR9331157  0.18849794  0.21166385  -0.08779447  0.03151171     -0.12345672</span></span>
<span><span class="co">#&gt; SRR9331158  0.18746348  0.06860765  -0.23436093 -0.22501425      0.08083409</span></span>
<span><span class="co">#&gt; SRR9331159  0.21133109  0.23461492   0.06919394  0.01037359     -0.22906818</span></span>
<span><span class="co">#&gt; SRR9331160  0.09663064  0.18920817  -0.04229932 -0.02042564     -0.06268189</span></span>
<span><span class="co">#&gt; SRR9331161 -0.19007306 -0.01012471   0.01179415  0.21475690      0.25418041</span></span>
<span><span class="co">#&gt;             MEhoneydew1   MEskyblue     MEbrown MElavenderblush3     MEgreen</span></span>
<span><span class="co">#&gt; SRR9331156  0.119562501 -0.01621926  0.19310610     -0.037796161  0.03519341</span></span>
<span><span class="co">#&gt; SRR9331157  0.112527169  0.12491312  0.03259990      0.043107418  0.04621808</span></span>
<span><span class="co">#&gt; SRR9331158 -0.003695732  0.19282362 -0.22194670     -0.069485541 -0.17835832</span></span>
<span><span class="co">#&gt; SRR9331159 -0.143880322 -0.11824264 -0.00550279     -0.002720895 -0.05812352</span></span>
<span><span class="co">#&gt; SRR9331160  0.089585237  0.17195936  0.04222481      0.098539105  0.11178514</span></span>
<span><span class="co">#&gt; SRR9331161  0.056873328  0.04899228  0.26705874      0.217998654  0.14229403</span></span>
<span><span class="co">#&gt;                  MEtan   MEdarkred       MEred       MEgrey</span></span>
<span><span class="co">#&gt; SRR9331156  0.08383924 -0.05607840 -0.04201499  0.006968271</span></span>
<span><span class="co">#&gt; SRR9331157  0.12610404 -0.05015207 -0.03727994  0.048408686</span></span>
<span><span class="co">#&gt; SRR9331158  0.05884285 -0.02513959 -0.11503723  0.012758211</span></span>
<span><span class="co">#&gt; SRR9331159 -0.16331102 -0.22126207 -0.05907953  0.011943965</span></span>
<span><span class="co">#&gt; SRR9331160  0.06474531 -0.16222775  0.02724823 -0.076827412</span></span>
<span><span class="co">#&gt; SRR9331161  0.21955979  0.01989451  0.05276451  0.122550745</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="calculating-expression-divergence">Calculating expression divergence<a class="anchor" aria-label="anchor" href="#calculating-expression-divergence"></a>
</h2>
<p><em><a href="https://github.com/exdiva" class="external-link">exdiva</a></em> offers
different metrics of expression divergence, namely:</p>
<ol style="list-style-type: decimal">
<li>Spearman correlations between gene pairs;</li>
<li>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
index of tissue specificity;</li>
<li>Co-occurrence in coexpression modules;</li>
<li>Distance between module eigengenes.</li>
</ol>
<div class="section level3">
<h3 id="metric-1-spearman-correlations-between-gene-pairs">Metric 1: Spearman correlations between gene pairs<a class="anchor" aria-label="anchor" href="#metric-1-spearman-correlations-between-gene-pairs"></a>
</h3>
<p>To calculate correlations between gene pairs, you will use the
function <code><a href="../reference/calculate_cor.html">calculate_cor()</a></code>, which takes as input an object of
class <code>SummarizedExperiment</code>,
<code>SingleCellExperiment</code>, or
<code>SpatialExperiment</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate Spearman correlations between gene pairs</span></span>
<span><span class="va">dup10</span> <span class="op">&lt;-</span> <span class="va">ath_dups</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span> <span class="co"># subset first 10 pairs for demonstration purposes</span></span>
<span><span class="va">cor_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_cor.html">calculate_cor</a></span><span class="op">(</span><span class="va">ath_se</span>, <span class="va">dup10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show output</span></span>
<span><span class="va">cor_df</span></span>
<span><span class="co">#&gt;         dup1      dup2 type         cor</span></span>
<span><span class="co">#&gt; 1  AT2G17930 AT4G36080   SD  0.65714286</span></span>
<span><span class="co">#&gt; 2  AT1G03060 AT4G02660   SD  0.94285714</span></span>
<span><span class="co">#&gt; 3  AT2G45540 AT3G60920   SD  0.25714286</span></span>
<span><span class="co">#&gt; 4  ATCG01280 ATCG00860   SD          NA</span></span>
<span><span class="co">#&gt; 5  AT3G28770 AT5G39870   SD          NA</span></span>
<span><span class="co">#&gt; 6  AT1G05570 AT2G31960   SD  0.31428571</span></span>
<span><span class="co">#&gt; 7  AT3G60860 AT1G01960   SD  0.71428571</span></span>
<span><span class="co">#&gt; 8  ATCG01130 ATCG01000   SD          NA</span></span>
<span><span class="co">#&gt; 9  AT2G33240 AT1G04600   SD -0.02857143</span></span>
<span><span class="co">#&gt; 10 AT1G79000 AT3G12980   SD  0.88571429</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="metric-2-the-tau-index-of-tissue-specificity">Metric 2: The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
index of tissue specificity<a class="anchor" aria-label="anchor" href="#metric-2-the-tau-index-of-tissue-specificity"></a>
</h3>
<p>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
index of tissue specificity ranges from 0 to 1, with values close to 0
indicating a broader expression profile across tissues, and values close
to 1 indicating a more tissue-specific profile. To calculate
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>,
we need a matrix of gene expression aggregated to tissue level, with
genes in rows and tissue names in columns. This can be obtained from a
<code>SummarizedExperiment</code> object with the function
<code><a href="../reference/aggregate_to_tissue.html">aggregate_to_tissue()</a></code>, which calculates the median (or
mean) gene expression per tissue using a user-defined variable in the
<code>colData</code> slot. However, since we have samples from a single
tissue only, we will simulate a two-tissue data set by modifying our
<code>colData</code> variable <strong>tissue</strong>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Change tissue name in half of the samples</span></span>
<span><span class="va">ath_se</span><span class="op">$</span><span class="va">tissue</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"leaf"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ath_se</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 3 columns</span></span>
<span><span class="co">#&gt;             experiment      tissue   treatment</span></span>
<span><span class="co">#&gt;            &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; SRR9659514   SRP213876        leaf         Red</span></span>
<span><span class="co">#&gt; SRR9659515   SRP213876        leaf         Red</span></span>
<span><span class="co">#&gt; SRR9659516   SRP213876        leaf         Red</span></span>
<span><span class="co">#&gt; SRR9659517   SRP213876       seeds     Far-red</span></span>
<span><span class="co">#&gt; SRR9659518   SRP213876       seeds     Far-red</span></span>
<span><span class="co">#&gt; SRR9659519   SRP213876       seeds     Far-red</span></span>
<span></span>
<span><span class="co"># Get median expression per tissue using "tissue" variable in colData</span></span>
<span><span class="va">aexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/aggregate_to_tissue.html">aggregate_to_tissue</a></span><span class="op">(</span><span class="va">ath_se</span>, <span class="st">"tissue"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">aexp</span><span class="op">)</span></span>
<span><span class="co">#&gt;                leaf     seeds</span></span>
<span><span class="co">#&gt; AT1G01010  76.01148  85.64420</span></span>
<span><span class="co">#&gt; AT1G01020 460.21158 303.03580</span></span>
<span><span class="co">#&gt; AT1G01030  46.13850  21.04727</span></span>
<span><span class="co">#&gt; AT1G01040 564.81950 687.85156</span></span>
<span><span class="co">#&gt; AT1G01050 605.36630 637.83575</span></span>
<span><span class="co">#&gt; AT1G01060 731.03310 123.45633</span></span></code></pre></div>
<p>Now that we have a matrix of gene expression per tissue, we can use
the function <code><a href="../reference/calculate_tau.html">calculate_tau()</a></code>. Besides, calculating
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>,
this function classifies genes into one of four expression-based
categories:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Null:</strong> Expression &lt;1 TPM in all tissues.</li>
<li>
<strong>Weak:</strong> Expression &lt;5 TPM in all tissues.</li>
<li>
<strong>Broad:</strong>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
&lt;0.85.</li>
<li>
<strong>Specific:</strong>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
&gt;= 0.85.</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate tau index of tissue specificity</span></span>
<span><span class="va">tau_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_tau.html">calculate_tau</a></span><span class="op">(</span><span class="va">aexp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tau_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;        gene         tau class</span></span>
<span><span class="co">#&gt; 1 AT1G01010 0.026414297 Broad</span></span>
<span><span class="co">#&gt; 2 AT1G01020 0.067936283 Broad</span></span>
<span><span class="co">#&gt; 3 AT1G01030 0.197218638 Broad</span></span>
<span><span class="co">#&gt; 4 AT1G01040 0.030107101 Broad</span></span>
<span><span class="co">#&gt; 5 AT1G01050 0.008075227 Broad</span></span>
<span><span class="co">#&gt; 6 AT1G01060 0.268635181 Broad</span></span></code></pre></div>
<p>Then, this data frame with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
indices per gene can be used to compare expression profiles of
duplicated gene pairs using the function <code><a href="../reference/compare_tau.html">compare_tau()</a></code> as
follows:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare tau indices among duplicated gene pairs</span></span>
<span><span class="va">tau_comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_tau.html">compare_tau</a></span><span class="op">(</span><span class="va">tau_df</span>, <span class="va">ath_dups</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tau_comp</span><span class="op">)</span></span>
<span><span class="co">#&gt;        dup1      dup2 type    tau_dup1 class_dup1   tau_dup2 class_dup2</span></span>
<span><span class="co">#&gt; 1 AT4G01550 AT1G01010   DD 0.058046902      Broad 0.02641430      Broad</span></span>
<span><span class="co">#&gt; 2 AT1G02220 AT1G01010   DD 0.404289662      Broad 0.02641430      Broad</span></span>
<span><span class="co">#&gt; 3 AT4G01510 AT1G01020   DD 0.162417251      Broad 0.06793628      Broad</span></span>
<span><span class="co">#&gt; 4 AT5G45150 AT1G01040   DD 0.003416496      Broad 0.03010710      Broad</span></span>
<span><span class="co">#&gt; 5 AT4G15417 AT1G01040   DD 1.000000000       Weak 0.03010710      Broad</span></span>
<span><span class="co">#&gt; 6 AT1G35530 AT1G01040   DD 0.143371321      Broad 0.03010710      Broad</span></span>
<span><span class="co">#&gt;     tau_diff class_preserved</span></span>
<span><span class="co">#&gt; 1 0.03163261            TRUE</span></span>
<span><span class="co">#&gt; 2 0.37787537            TRUE</span></span>
<span><span class="co">#&gt; 3 0.09448097            TRUE</span></span>
<span><span class="co">#&gt; 4 0.02669061            TRUE</span></span>
<span><span class="co">#&gt; 5 0.96989290           FALSE</span></span>
<span><span class="co">#&gt; 6 0.11326422            TRUE</span></span></code></pre></div>
<p>The column <code>tau_diff</code> indicates the absolute difference in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
indices between gene pairs, and the column <code>class_preserved</code>
indicates whether expression-based classes are the same in both members
of a gene pair.</p>
</div>
<div class="section level3">
<h3 id="metric-3-co-occurrence-in-coexpression-modules">Metric 3: Co-occurrence in coexpression modules<a class="anchor" aria-label="anchor" href="#metric-3-co-occurrence-in-coexpression-modules"></a>
</h3>
<p>Gene coexpression networks can be used to identify cluster of genes
with very similar expression profiles across samples, which are termed
<strong>coexpression modules</strong>. Here, <em><a href="https://github.com/exdiva" class="external-link">exdiva</a></em> uses such information
to compare module assignment for genes in a pair and check if they are
in the same module (and thus, have similar expression profiles) or in
different modules. This can be done with the function
<code><a href="../reference/compare_coex_modules.html">compare_coex_modules()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare module assignment between duplicate pairs</span></span>
<span><span class="va">mod_comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_coex_modules.html">compare_coex_modules</a></span><span class="op">(</span><span class="va">ath_dups</span>, <span class="va">genes_and_modules</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mod_comp</span><span class="op">)</span></span>
<span><span class="co">#&gt;        dup1      dup2 type module_dup1 module_dup2 module_preservation</span></span>
<span><span class="co">#&gt; 1 AT1G02220 AT1G01010   DD        &lt;NA&gt;        &lt;NA&gt;         both_absent</span></span>
<span><span class="co">#&gt; 2 AT4G01550 AT1G01010   DD        &lt;NA&gt;        &lt;NA&gt;         both_absent</span></span>
<span><span class="co">#&gt; 3 AT4G01510 AT1G01020   DD        &lt;NA&gt;        &lt;NA&gt;         both_absent</span></span>
<span><span class="co">#&gt; 4 AT4G15417 AT1G01040   DD        &lt;NA&gt;      orange          one_absent</span></span>
<span><span class="co">#&gt; 5 AT1G35530 AT1G01040   DD        &lt;NA&gt;      orange          one_absent</span></span>
<span><span class="co">#&gt; 6 AT5G45150 AT1G01040   DD        &lt;NA&gt;      orange          one_absent</span></span></code></pre></div>
<p>The variable <code>module_preservation</code> summarizes results for
each gene pair in a factor with the following levels:</p>
<ol style="list-style-type: decimal">
<li>
<strong>preserved:</strong> Both genes are in the same module.</li>
<li>
<strong>diverged:</strong> Genes are in different modules.</li>
<li>
<strong>one_absent:</strong> One gene in the pair is not present in
any module. This could be because the gene is not present in the input
expression data (due to low or no expression), or because the module
detection algorithm could not assign the gene to any module.</li>
<li>
<strong>both_absent:</strong> Both genes are not present in any
module.</li>
</ol>
<p>To get the frequency of gene pairs in each category, you can use:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get frequency of gene pairs for each category</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">mod_comp</span><span class="op">$</span><span class="va">module_preservation</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   preserved    diverged  one_absent both_absent </span></span>
<span><span class="co">#&gt;         998        2924       13223       30340</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="metric-4-distance-between-module-eigengenes">Metric 4: Distance between module eigengenes<a class="anchor" aria-label="anchor" href="#metric-4-distance-between-module-eigengenes"></a>
</h3>
<p>Co-occurrence in coexpression modules is binary: genes are either in
the same module or not. However, genes can be in different, but
(somewhat) similar modules, or in completely different modules. In that
case, we need a way of comparing distance between modules. Here, we do
so by calculating correlations between module eigengenes (i.e., the
first principal component of each module, which describes most of the
variation). This can be done with the function
<code><a href="../reference/compare_coex_me.html">compare_coex_me()</a></code>. Importantly, such comparison is only
made for genes in the <strong>diverged</strong> category (see previous
section for details).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare module eigengenes for diverged duplicate pairs</span></span>
<span><span class="va">me_comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_coex_me.html">compare_coex_me</a></span><span class="op">(</span><span class="va">mod_comp</span>, <span class="va">ME</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">me_comp</span><span class="op">)</span></span>
<span><span class="co">#&gt;        dup1      dup2 type      ME_cor</span></span>
<span><span class="co">#&gt; 1 AT2G24270 AT1G79440   DD  0.06105458</span></span>
<span><span class="co">#&gt; 2 AT2G33990 AT5G03040 dTRD -0.56880204</span></span>
<span><span class="co">#&gt; 3 AT2G38400 AT5G46180   DD -0.56880204</span></span>
<span><span class="co">#&gt; 4 AT5G55700 AT3G23920   DD -0.56880204</span></span>
<span><span class="co">#&gt; 5 AT4G39660 AT1G80600   DD -0.56880204</span></span>
<span><span class="co">#&gt; 6 AT2G33990 AT3G22190 dTRD -0.56880204</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-08-26</span></span>
<span><span class="co">#&gt;  pandoc   3.2 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version</span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-5   <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase              * 2.64.0  <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.50.0  <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.24 <span style="color: #555555;">2024-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.32.1  <span style="color: #555555;">2024-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.40    <span style="color: #555555;">2024-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.8.0   <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.1.0   <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.3   <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.3   <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.30.1  <span style="color: #555555;">2024-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.3   <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.37  <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  evaluate               0.24.0  <span style="color: #555555;">2024-06-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  exdiva               * 0.99.0  <span style="color: #555555;">2024-08-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  fastmap                1.2.0   <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.4   <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.40.1  <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.12  <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.56.1  <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1 <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4   <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr                   1.4.7   <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.38.1  <span style="color: #555555;">2024-07-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4   <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.8   <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.48    <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.22-6  <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4   <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.7-0   <span style="color: #555555;">2024-04-26</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.16.0  <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  matrixStats          * 1.3.0   <span style="color: #555555;">2024-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                2.1.0   <span style="color: #555555;">2024-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1   <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.3.2   <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.4   <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.28    <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.4.1   <span style="color: #555555;">2024-05-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.42.1  <span style="color: #555555;">2024-07-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.9   <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo            1.2.2   <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SparseArray            1.4.8   <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.34.0  <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.1.0   <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.4.0   <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils             1.0.0   <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.47    <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.44.0  <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.10  <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.50.0  <span style="color: #555555;">2024-04-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.19 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-almeida2024doubletrouble" class="csl-entry">
Almeida-Silva, Fabricio, and Yves Van de Peer. 2024.
<span>“Doubletrouble: An r/Bioconductor Package for the Identification,
Classification, and Analysis of Gene and Genome Duplications.”</span>
<em>bioRxiv</em>, 2024–02.
</div>
<div id="ref-tognacca2019alternative" class="csl-entry">
Tognacca, Rocı́o Soledad, Lucas Servi, Carlos Esteban Hernando, Maite
Saura-Sanchez, Marcelo Javier Yanovsky, Ezequiel Petrillo, and Javier
Francisco Botto. 2019. <span>“Alternative Splicing Regulation During
Light-Induced Germination of Arabidopsis Thaliana Seeds.”</span>
<em>Frontiers in Plant Science</em> 10: 468265.
</div>
<div id="ref-wan2019comparing" class="csl-entry">
Wan, Wei-Lin, Lisha Zhang, Rory Pruitt, Maricris Zaidem, Rik Brugman,
Xiyu Ma, Elzbieta Krol, et al. 2019. <span>“Comparing Arabidopsis
Receptor Kinase and Receptor Protein-Mediated Immune Signaling Reveals
BIK1-Dependent Differences.”</span> <em>New Phytologist</em> 221 (4):
2080–95.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
